### ISSUE TITLE
On the use of 'gt_mask'

### ISSUE THREAD

**Issue by carlvinson70** [Cc-Hy/CMKD#33]
Hi, could you please explain how 'gt_mask' can be utilised? Apparently it shows up in the cmkd.py script and doesn't seem to be used in training. Many thanks.

**Comment by Cc-Hy** [comment]
Hi, @carlvinson70 
The gt_mask is used earlier in the framework, and we remove it later because we suppose that our framework is trained with unlabeled data and the gt labels are not available. 
If you want to use it, you can only use it with the labeled samples, specifically, you can generate a binary mask in the BEV space to indicate whether a BEV position is occupied by foreground objects so that you can use a larger loss weight. But I remembered this does not give an improvement in performance, you can have a try if you are interested.

**Comment by carlvinson70** [comment]
Thanks for the response. Actually, when training with both labelled + unlabelled data, how can you use the labels of the labelled data to supervise the labelled data (on-the-fly)? - I think this is more intuitive but am not sure how the existing code may be modified for this.

**Comment by carlvinson70** [comment]
Thanks for the info. @Cc-Hy 

Another quick question: is the IoU confidence score weighting (that is, 's' in equation 4 & 5) implemented in the current code?

**Comment by Cc-Hy** [comment]
@carlvinson70 
Hi, 's' is implemented for the regression loss, but not the classification loss in the current code. See [this line](https://github.com/Cc-Hy/CMKD/blob/main/pcdet/models/dense_heads/anchor_head_single_cmkd_v2.py#L123).

**Comment by carlvinson70** [comment]
> @carlvinson70 Hi, 's' is implemented for the regression loss, but not the classification loss in the current code. See [this line](https://github.com/Cc-Hy/CMKD/blob/main/pcdet/models/dense_heads/anchor_head_single_cmkd_v2.py#L123).

From what I understand from this line, it is using the sum of "logits" of all 3 classes as a confidence score? This seems different from the descriptions in the paper, if I understand it right? And how is this confidence IoU-related?

**Comment by Cc-Hy** [comment]
's' here is the predicted classification confidence score of the LiDAR-based teacher model. When training the teacher model, we use the quality labels like IoU or centerness as the classification ground truth, so the predicted confidence scores of the soft labels are IoU-related or centerness-related which are used to weight the losses.
